### DISCUSSION TITLE
Add support for custom datasources without `defaultRegistryUrlTemplate`

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#27148] [Suggest an Idea]
### Tell us more.

JSONata can create objects even if there is no input document, for example: https://try.jsonata.org/1o1fwosgk
Those kind of custom datasources could be more efficient and less error-prone if it could optionally skip reading a file or fetching from a URL.

**Comment by secustor** [comment]
I have nothing against making `defaultRegistryUrl` optional, but I'm not sure if this would make it easier for users. The data would rather have to be in an escaped single line of string, rather than an external pretty printed file in the repository.  

What is your exact usecase here?  
Do you want to provide a 'static' datasource in an organization?

**Reply by astellingwerf** [reply]
In my case, it's close, but not 100% static. It should return a version number with the two last digits of the year, and the quarter, both as if the date is actually three weeks from today. So, last Christmas (🎵), it would have returned `{releases:[{version: "24.1.0"}]}`.

**Reply by secustor** [reply]
This should be a different `format` like `no-input` to make it easier understanding what the purpose is here.

**Reply by astellingwerf** [reply]
Makes sense!

**Reply by astellingwerf** [reply]
@secustor, can you create an issue from this discussion?

**Comment by secustor** [comment]
https://github.com/renovatebot/renovate/issues/31862
